### ISSUE TITLE
set-source-date-epoch-to-latest.sh uses non-portable 'stat -c'

### ISSUE THREAD

**Issue by scode** [NixOS/nixpkgs#12809]
(Observed on OS X 10.11.3)

For at least a couple of ruby packages (not sure the extent), I noticed:

```
unpacking sources
find: cannot search `': No such file or directory
/nix/store/0p1afvl8jcpi6dvsq2n58i90w9c59vz1-set-source-date-epoch-to-latest.sh: line 12: [: : integer expression expected
```

This is coming from `./pkgs/build-support/setup-hooks/set-source-date-epoch-to-latest.sh` which is doing this:

```
    # Get the last modification time of all regular files, sort them,
    # and get the most recent. Maybe we should use
    # https://github.com/0-wiz-0/findnewest here.
    local -a res=($(find "$path" -type f -print0 | xargs -0 -r stat -c '%Y %n' | sort -n | tail -n1))
    local time="${res[0]}"
    local newestFile="${res[1]}"

    # Update $SOURCE_DATE_EPOCH if the most recent file we found is newer.
    if [ "$time" -gt "$SOURCE_DATE_EPOCH" ]; then
```

And the error is because `$time` evaluates to the empty string.

Not sure yet if there is another problem, but at minimum, `stat -c` is not supported on OS X. `stat -f` is, but that means something entirely different on GNU systems.

Example repro: `nix-env -i lolcat`

**Comment by scode** [comment]
(Considering solutions. Didn't realize until after I submitted that I can't assign to myself.)

**Comment by scode** [comment]
Given that $path is non-existent, the actual error above is irrelevant to the problem at hand since stat is never executed for an empty set of files; i.e. the included log output is a red herring and mostly just represents how I ended up realizing this was a problem.

**Comment by scode** [comment]
The initial version landed in 81e530a7499e3f06bec13b6ec974d0f1e3a4ab84 and contained a check for darwin. In 38460cfe7246f34ccb09768b1435b9dcfd739e3b the check was removed, but I don't think the actual implementation was adjusted to actually support darwin?

**Comment by scode** [comment]
Okay. Reading up on https://github.com/NixOS/nixpkgs/commit/81e530a7499e3f06bec13b6ec974d0f1e3a4ab84 it seems this is my bad. Since the toolchain used in nix is going to have the appropriate version of 'stat', this entire issue is a red herring and isn't a problem. Hence the removal of the check.

Apologies for the noise generated - closing.
